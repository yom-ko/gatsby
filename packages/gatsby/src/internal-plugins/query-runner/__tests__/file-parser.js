jest.mock(`fs-extra`, () => {
  let mockFiles = {}
  return {
    __setMockFiles: newMockFiles => {
      mockFiles = Object.assign({}, newMockFiles)
    },
    readFile: (filePath, parser) =>
      new Promise(resolve => resolve(mockFiles[filePath])),
  }
})
jest.mock(`../../../utils/api-runner-node`, () => () => [])

const FileParser = require(`../file-parser`).default

describe(`File parser`, () => {
  const MOCK_FILE_INFO = {
    "no-query.js": `import React from "react"`,
    "page-query.js": `import { graphql } from 'gatsby'
    export const query = graphql\`
query PageQueryName {
  foo
}
\``,
    "page-query-no-name.js": `import { graphql } from 'gatsby'
  export const query = graphql\`
query {
  foo
}
\``,
    "static-query.js": `import { graphql } from 'gatsby'
  export default () => (
  <StaticQuery
    query={graphql\`query StaticQueryName { foo }\`}
    render={data => <div>{data.doo}</div>}
  />
)`,
    "static-query-no-name.js": `import { graphql } from 'gatsby'
  export default () => (
  <StaticQuery
    query={graphql\`{ foo }\`}
    render={data => <div>{data.foo}</div>}
  />
)`,
    "static-query-named-export.js": `import { graphql } from 'gatsby'
  export const Component = () => (
  <StaticQuery
    query={graphql\`query StaticQueryName { foo }\`}
    render={data => <div>{data.doo}</div>}
  />
)`,
    "static-query-closing-tag.js": `import { graphql } from 'gatsby'
  export default () => (
  <StaticQuery
    query={graphql\`{ foo }\`}
  >
    {data => <div>{data.foo}</div>}
  </StaticQuery>
)`,
    "page-query-and-static-query-named-export.js": `import { graphql } from 'gatsby'
  export const Component = () => (
  <StaticQuery
    query={graphql\`query StaticQueryName { foo }\`}
    render={data => <div>{data.doo}</div>}
  />
)
export const pageQuery = graphql\`query PageQueryName { foo }\`
`,
    "multiple-fragment-exports.js": `import { graphql } from 'gatsby'
  export const fragment1 = graphql\`
  fragment Fragment1 on RootQueryField {
    foo
  }
  fragment Fragment2 on RootQueryField {
    bar
  }
\`
export const fragment3 = graphql\`
  fragment Fragment3 on RootQueryField {
    baz
  }
\`
`,
    "fragment-shorthand.js": `import React from "react"
import { StaticQuery, graphql } from "gatsby"

const query = graphql\`
  query {
    site {
      siteMetadata {
        title
      }
    }
  }
\`

export default () => (
  <>
    <StaticQuery
      query={query}
      render={data => <div>{data.title}</div>}
    />
  </>
)`,
    "query-in-separate-variable.js": `import React from "react"
import { StaticQuery, graphql } from "gatsby"

const query = graphql\`{ allMarkdownRemark { blah { node { cheese }}}}\`

export default () => (
  <StaticQuery
    query={query}
    render={data => <div>{data.pizza}</div>}
  />
)`,
    "query-in-separate-variable-2.js": `import React from "react"
import { StaticQuery, graphql } from "gatsby"

const query = graphql\`{ fakeOut { blah { node { cheese }}}}\`
const strangeQueryName = graphql\`{ allStrangeQueryName { blah { node { cheese }}}}\`

export default () => (
  <StaticQuery
    query={strangeQueryName}
    render={data => <div>{data.pizza}</div>}
  />
)`,
    "query-not-defined.js": `import React from "react"
import { StaticQuery, graphql } from "gatsby"

export default () => (
  <StaticQuery
    query={strangeQueryName}
    render={data => <div>{data.pizza}</div>}
  />
)`,
    "query-imported.js": `import React from "react"
import { StaticQuery, graphql } from "gatsby"
import strangeQueryName from "./another-file.js"

export default () => (
  <StaticQuery
    query={strangeQueryName}
    render={data => <div>{data.pizza}</div>}
  />
)`,
  }

  const parser = new FileParser()

  beforeAll(() => {
    require(`fs-extra`).__setMockFiles(MOCK_FILE_INFO)
  })

  it(`extracts query AST correctly from files`, async () => {
    const spyStderr = jest.spyOn(process.stderr, `write`)
    const results = await parser.parseFiles(Object.keys(MOCK_FILE_INFO))
    expect(results).toMatchSnapshot()
    expect(
      spyStderr.mock.calls
        .filter(c => c[0].includes(`warning`))
        // Remove console colors + trim whitespace
        // eslint-disable-next-line
        .map(c => c[0].replace(/\x1B[[(?);]{0,2}(;?\d)*./g, ``).trim())
    ).toMatchSnapshot()
    spyStderr.mockRestore()
  })
})
