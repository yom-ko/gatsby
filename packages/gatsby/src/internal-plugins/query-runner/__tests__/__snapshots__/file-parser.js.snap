// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File parser extracts query AST correctly from files 1`] = `
Map {
  "page-query.js" => Object {
    "definitions": Array [
      Object {
        "directives": Array [],
        "kind": "OperationDefinition",
        "loc": Object {
          "end": 30,
          "start": 1,
        },
        "name": Object {
          "kind": "Name",
          "loc": Object {
            "end": 20,
            "start": 7,
          },
          "value": "PageQueryName",
        },
        "operation": "query",
        "selectionSet": Object {
          "kind": "SelectionSet",
          "loc": Object {
            "end": 30,
            "start": 21,
          },
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 28,
                "start": 25,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 28,
                  "start": 25,
                },
                "value": "foo",
              },
              "selectionSet": undefined,
            },
          ],
        },
        "variableDefinitions": Array [],
      },
    ],
    "kind": "Document",
  },
  "page-query-no-name.js" => Object {
    "definitions": Array [
      Object {
        "directives": Array [],
        "kind": "OperationDefinition",
        "loc": Object {
          "end": 16,
          "start": 1,
        },
        "name": Object {
          "kind": "Name",
          "value": "pageQueryNoNameJs2538359280",
        },
        "operation": "query",
        "selectionSet": Object {
          "kind": "SelectionSet",
          "loc": Object {
            "end": 16,
            "start": 7,
          },
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 14,
                "start": 11,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 14,
                  "start": 11,
                },
                "value": "foo",
              },
              "selectionSet": undefined,
            },
          ],
        },
        "variableDefinitions": Array [],
      },
    ],
    "kind": "Document",
  },
  "static-query.js" => Object {
    "definitions": Array [
      Object {
        "directives": Array [],
        "hash": 407706182,
        "isStaticQuery": true,
        "kind": "OperationDefinition",
        "loc": Object {
          "end": 29,
          "start": 0,
        },
        "name": Object {
          "kind": "Name",
          "loc": Object {
            "end": 21,
            "start": 6,
          },
          "value": "StaticQueryName",
        },
        "operation": "query",
        "selectionSet": Object {
          "kind": "SelectionSet",
          "loc": Object {
            "end": 29,
            "start": 22,
          },
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 27,
                "start": 24,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 27,
                  "start": 24,
                },
                "value": "foo",
              },
              "selectionSet": undefined,
            },
          ],
        },
        "text": "query StaticQueryName { foo }",
        "variableDefinitions": Array [],
      },
    ],
    "kind": "Document",
  },
  "static-query-no-name.js" => Object {
    "definitions": Array [
      Object {
        "directives": Array [],
        "hash": 2605331189,
        "isStaticQuery": true,
        "kind": "OperationDefinition",
        "loc": Object {
          "end": 7,
          "start": 0,
        },
        "name": Object {
          "kind": "Name",
          "value": "staticQueryNoNameJs2605331189",
        },
        "operation": "query",
        "selectionSet": Object {
          "kind": "SelectionSet",
          "loc": Object {
            "end": 7,
            "start": 0,
          },
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 5,
                "start": 2,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 5,
                  "start": 2,
                },
                "value": "foo",
              },
              "selectionSet": undefined,
            },
          ],
        },
        "text": "{ foo }",
        "variableDefinitions": Array [],
      },
    ],
    "kind": "Document",
  },
  "static-query-named-export.js" => Object {
    "definitions": Array [
      Object {
        "directives": Array [],
        "hash": 407706182,
        "isStaticQuery": true,
        "kind": "OperationDefinition",
        "loc": Object {
          "end": 29,
          "start": 0,
        },
        "name": Object {
          "kind": "Name",
          "loc": Object {
            "end": 21,
            "start": 6,
          },
          "value": "StaticQueryName",
        },
        "operation": "query",
        "selectionSet": Object {
          "kind": "SelectionSet",
          "loc": Object {
            "end": 29,
            "start": 22,
          },
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 27,
                "start": 24,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 27,
                  "start": 24,
                },
                "value": "foo",
              },
              "selectionSet": undefined,
            },
          ],
        },
        "text": "query StaticQueryName { foo }",
        "variableDefinitions": Array [],
      },
    ],
    "kind": "Document",
  },
  "static-query-closing-tag.js" => Object {
    "definitions": Array [
      Object {
        "directives": Array [],
        "hash": 2605331189,
        "isStaticQuery": true,
        "kind": "OperationDefinition",
        "loc": Object {
          "end": 7,
          "start": 0,
        },
        "name": Object {
          "kind": "Name",
          "value": "staticQueryClosingTagJs2605331189",
        },
        "operation": "query",
        "selectionSet": Object {
          "kind": "SelectionSet",
          "loc": Object {
            "end": 7,
            "start": 0,
          },
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 5,
                "start": 2,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 5,
                  "start": 2,
                },
                "value": "foo",
              },
              "selectionSet": undefined,
            },
          ],
        },
        "text": "{ foo }",
        "variableDefinitions": Array [],
      },
    ],
    "kind": "Document",
  },
  "page-query-and-static-query-named-export.js" => Object {
    "definitions": Array [
      Object {
        "directives": Array [],
        "hash": 407706182,
        "isStaticQuery": true,
        "kind": "OperationDefinition",
        "loc": Object {
          "end": 29,
          "start": 0,
        },
        "name": Object {
          "kind": "Name",
          "loc": Object {
            "end": 21,
            "start": 6,
          },
          "value": "StaticQueryName",
        },
        "operation": "query",
        "selectionSet": Object {
          "kind": "SelectionSet",
          "loc": Object {
            "end": 29,
            "start": 22,
          },
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 27,
                "start": 24,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 27,
                  "start": 24,
                },
                "value": "foo",
              },
              "selectionSet": undefined,
            },
          ],
        },
        "text": "query StaticQueryName { foo }",
        "variableDefinitions": Array [],
      },
      Object {
        "directives": Array [],
        "kind": "OperationDefinition",
        "loc": Object {
          "end": 27,
          "start": 0,
        },
        "name": Object {
          "kind": "Name",
          "loc": Object {
            "end": 19,
            "start": 6,
          },
          "value": "PageQueryName",
        },
        "operation": "query",
        "selectionSet": Object {
          "kind": "SelectionSet",
          "loc": Object {
            "end": 27,
            "start": 20,
          },
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 25,
                "start": 22,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 25,
                  "start": 22,
                },
                "value": "foo",
              },
              "selectionSet": undefined,
            },
          ],
        },
        "variableDefinitions": Array [],
      },
    ],
    "kind": "Document",
  },
  "multiple-fragment-exports.js" => Object {
    "definitions": Array [
      Object {
        "directives": Array [],
        "kind": "FragmentDefinition",
        "loc": Object {
          "end": 53,
          "start": 3,
        },
        "name": Object {
          "kind": "Name",
          "loc": Object {
            "end": 21,
            "start": 12,
          },
          "value": "Fragment1",
        },
        "selectionSet": Object {
          "kind": "SelectionSet",
          "loc": Object {
            "end": 53,
            "start": 40,
          },
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 49,
                "start": 46,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 49,
                  "start": 46,
                },
                "value": "foo",
              },
              "selectionSet": undefined,
            },
          ],
        },
        "typeCondition": Object {
          "kind": "NamedType",
          "loc": Object {
            "end": 39,
            "start": 25,
          },
          "name": Object {
            "kind": "Name",
            "loc": Object {
              "end": 39,
              "start": 25,
            },
            "value": "RootQueryField",
          },
        },
      },
      Object {
        "directives": Array [],
        "kind": "FragmentDefinition",
        "loc": Object {
          "end": 106,
          "start": 56,
        },
        "name": Object {
          "kind": "Name",
          "loc": Object {
            "end": 74,
            "start": 65,
          },
          "value": "Fragment2",
        },
        "selectionSet": Object {
          "kind": "SelectionSet",
          "loc": Object {
            "end": 106,
            "start": 93,
          },
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 102,
                "start": 99,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 102,
                  "start": 99,
                },
                "value": "bar",
              },
              "selectionSet": undefined,
            },
          ],
        },
        "typeCondition": Object {
          "kind": "NamedType",
          "loc": Object {
            "end": 92,
            "start": 78,
          },
          "name": Object {
            "kind": "Name",
            "loc": Object {
              "end": 92,
              "start": 78,
            },
            "value": "RootQueryField",
          },
        },
      },
      Object {
        "directives": Array [],
        "kind": "FragmentDefinition",
        "loc": Object {
          "end": 53,
          "start": 3,
        },
        "name": Object {
          "kind": "Name",
          "loc": Object {
            "end": 21,
            "start": 12,
          },
          "value": "Fragment3",
        },
        "selectionSet": Object {
          "kind": "SelectionSet",
          "loc": Object {
            "end": 53,
            "start": 40,
          },
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 49,
                "start": 46,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 49,
                  "start": 46,
                },
                "value": "baz",
              },
              "selectionSet": undefined,
            },
          ],
        },
        "typeCondition": Object {
          "kind": "NamedType",
          "loc": Object {
            "end": 39,
            "start": 25,
          },
          "name": Object {
            "kind": "Name",
            "loc": Object {
              "end": 39,
              "start": 25,
            },
            "value": "RootQueryField",
          },
        },
      },
    ],
    "kind": "Document",
  },
  "fragment-shorthand.js" => Object {
    "definitions": Array [
      Object {
        "directives": Array [],
        "hash": 1097489062,
        "isStaticQuery": true,
        "kind": "OperationDefinition",
        "loc": Object {
          "end": 74,
          "start": 3,
        },
        "name": Object {
          "kind": "Name",
          "value": "fragmentShorthandJs1097489062",
        },
        "operation": "query",
        "selectionSet": Object {
          "kind": "SelectionSet",
          "loc": Object {
            "end": 74,
            "start": 9,
          },
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 70,
                "start": 15,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 19,
                  "start": 15,
                },
                "value": "site",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "loc": Object {
                  "end": 70,
                  "start": 20,
                },
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "loc": Object {
                      "end": 64,
                      "start": 28,
                    },
                    "name": Object {
                      "kind": "Name",
                      "loc": Object {
                        "end": 40,
                        "start": 28,
                      },
                      "value": "siteMetadata",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "loc": Object {
                        "end": 64,
                        "start": 41,
                      },
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "loc": Object {
                            "end": 56,
                            "start": 51,
                          },
                          "name": Object {
                            "kind": "Name",
                            "loc": Object {
                              "end": 56,
                              "start": 51,
                            },
                            "value": "title",
                          },
                          "selectionSet": undefined,
                        },
                      ],
                    },
                  },
                ],
              },
            },
          ],
        },
        "text": "
  query {
    site {
      siteMetadata {
        title
      }
    }
  }
",
        "variableDefinitions": Array [],
      },
    ],
    "kind": "Document",
  },
  "query-in-separate-variable.js" => Object {
    "definitions": Array [
      Object {
        "directives": Array [],
        "hash": 3748317405,
        "isStaticQuery": true,
        "kind": "OperationDefinition",
        "loc": Object {
          "end": 47,
          "start": 0,
        },
        "name": Object {
          "kind": "Name",
          "value": "queryInSeparateVariableJs3748317405",
        },
        "operation": "query",
        "selectionSet": Object {
          "kind": "SelectionSet",
          "loc": Object {
            "end": 47,
            "start": 0,
          },
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 46,
                "start": 2,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 19,
                  "start": 2,
                },
                "value": "allMarkdownRemark",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "loc": Object {
                  "end": 46,
                  "start": 20,
                },
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "loc": Object {
                      "end": 45,
                      "start": 22,
                    },
                    "name": Object {
                      "kind": "Name",
                      "loc": Object {
                        "end": 26,
                        "start": 22,
                      },
                      "value": "blah",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "loc": Object {
                        "end": 45,
                        "start": 27,
                      },
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "loc": Object {
                            "end": 44,
                            "start": 29,
                          },
                          "name": Object {
                            "kind": "Name",
                            "loc": Object {
                              "end": 33,
                              "start": 29,
                            },
                            "value": "node",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "loc": Object {
                              "end": 44,
                              "start": 34,
                            },
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "loc": Object {
                                  "end": 42,
                                  "start": 36,
                                },
                                "name": Object {
                                  "kind": "Name",
                                  "loc": Object {
                                    "end": 42,
                                    "start": 36,
                                  },
                                  "value": "cheese",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
            },
          ],
        },
        "text": "{ allMarkdownRemark { blah { node { cheese }}}}",
        "variableDefinitions": Array [],
      },
    ],
    "kind": "Document",
  },
  "query-in-separate-variable-2.js" => Object {
    "definitions": Array [
      Object {
        "directives": Array [],
        "hash": 1352988887,
        "isStaticQuery": true,
        "kind": "OperationDefinition",
        "loc": Object {
          "end": 49,
          "start": 0,
        },
        "name": Object {
          "kind": "Name",
          "value": "queryInSeparateVariable2Js1352988887",
        },
        "operation": "query",
        "selectionSet": Object {
          "kind": "SelectionSet",
          "loc": Object {
            "end": 49,
            "start": 0,
          },
          "selections": Array [
            Object {
              "alias": undefined,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 48,
                "start": 2,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 21,
                  "start": 2,
                },
                "value": "allStrangeQueryName",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "loc": Object {
                  "end": 48,
                  "start": 22,
                },
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "loc": Object {
                      "end": 47,
                      "start": 24,
                    },
                    "name": Object {
                      "kind": "Name",
                      "loc": Object {
                        "end": 28,
                        "start": 24,
                      },
                      "value": "blah",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "loc": Object {
                        "end": 47,
                        "start": 29,
                      },
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "loc": Object {
                            "end": 46,
                            "start": 31,
                          },
                          "name": Object {
                            "kind": "Name",
                            "loc": Object {
                              "end": 35,
                              "start": 31,
                            },
                            "value": "node",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "loc": Object {
                              "end": 46,
                              "start": 36,
                            },
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "loc": Object {
                                  "end": 44,
                                  "start": 38,
                                },
                                "name": Object {
                                  "kind": "Name",
                                  "loc": Object {
                                    "end": 44,
                                    "start": 38,
                                  },
                                  "value": "cheese",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
            },
          ],
        },
        "text": "{ allStrangeQueryName { blah { node { cheese }}}}",
        "variableDefinitions": Array [],
      },
    ],
    "kind": "Document",
  },
}
`;

exports[`File parser extracts query AST correctly from files 2`] = `
Array [
  "warning 
We were unable to find the declaration of variable \\"strangeQueryName\\", which you passed as the \\"query\\" prop into the <StaticQuery> declaration in \\"query-not-defined.js\\".

Perhaps the variable name has a typo?

Also note that we are currently unable to use queries defined in files other than the file where the <StaticQuery> is defined. If you're attempting to import the query, please move it into \\"query-not-defined.js\\". If being able to import queries from another file is an important capability for you, we invite your help fixing it.",
  "warning 
We were unable to find the declaration of variable \\"strangeQueryName\\", which you passed as the \\"query\\" prop into the <StaticQuery> declaration in \\"query-imported.js\\".

Perhaps the variable name has a typo?

Also note that we are currently unable to use queries defined in files other than the file where the <StaticQuery> is defined. If you're attempting to import the query, please move it into \\"query-imported.js\\". If being able to import queries from another file is an important capability for you, we invite your help fixing it.",
]
`;
